libfdisk/src/script.c

```diff
@@ -68,8 +68,42 @@ struct fdisk_script {
 
 	unsigned int		json : 1,		/* JSON output */
 				force_label : 1;	/* label: <name> specified */
+	unsigned int		no_device_names : 1;	/* 1=suppress device name, 0=display device name (default) */
 };
 
+/**
+ * fdisk_script_set_no_device_names - Set whether to display device names in script output
+ * @dp: script context
+ * @no: 1=suppress device name, 0=display device name
+ * After setting this flag, the script output will use partition numbers (such as "1:") 
+ * instead of device names (such as "/dev/sda1:"); 
+ * This makes the script more concise, easy to read, and suitable for cross-device use.
+ * 
```

Comment:
From this documentation it is not clear if only partition names should be affected or also the "device" field itself.